addons/point_of_sale/tests/test_pos_products_with_tax.py

```diff
@@ -294,6 +296,131 @@ def test_return_order(self):
+            category=self.categ_basic,
+            lst_price=12.10,
+            tax_ids=tax_20_incl.ids,
+        )
+        product2 = self.create_product(
+            name='Product 2',
+            category=self.categ_basic,
+            lst_price=6.05,
+            tax_ids=tax_20_incl.ids,
+        )
+        # Open a POS session without creating corresponding moves
+        self.open_new_session()
+        self.env['pos.order'].create_from_ui([self.create_ui_order_data([
+            (product1, 1),
+            (product2, -1),
+        ])])
+
+        # Create an entry
+        with Form(self.env["account.move"].with_context(
+                default_move_type="entry")) as move_form:
+            with move_form.line_ids.new() as line_form:
+                line_form.account_id = self.sale_account
+                # line_form.tax_ids = tax_21_incl
+                line_form.credit = 50.0
+                line_form.tax_ids.add(tax_20_incl)
+            with move_form.line_ids.new() as line_form_2:
+                line_form_2.account_id = self.pos_receivable_account
+                # line_form.tax_ids = tax_21_incl
+                line_form_2.debit = 50.0
+                line_form_2.tax_ids.add(tax_20_incl)
+        move = move_form.save()
+        # Check if tags are not affected by opened POS session
+        self.assertEqual(
+            move.line_ids.filtered(
+                lambda line: line.account_id == self.sale_account).tax_tag_ids,
+            base_20_tag_minus
+        )
+        self.assertEqual(
+            move.line_ids.filtered(
+                lambda line: line.account_id == self.tax_received_account).tax_tag_ids,
+            tax_20_tag_minus
+        )
+
+        # Close POS session
+        self.pos_session.action_pos_session_validate()
+
+        # Create an entry - check if the result is the same as above
+        with Form(self.env["account.move"].with_context(
+                default_move_type="entry")) as move_form:
+            with move_form.line_ids.new() as line_form:
+                line_form.account_id = self.sale_account
+                # line_form.tax_ids = tax_21_incl
+                line_form.credit = 50.0
+                line_form.tax_ids.add(tax_20_incl)
+            with move_form.line_ids.new() as line_form_2:
+                line_form_2.account_id = self.pos_receivable_account
+                # line_form.tax_ids = tax_21_incl
+                line_form_2.debit = 50.0
+                line_form_2.tax_ids.add(tax_20_incl)
+        move = move_form.save()
+        # Check if tags are not affected
+        self.assertEqual(
+            move.line_ids.filtered(
+                lambda line: line.account_id == self.sale_account).tax_tag_ids,
+            base_20_tag_minus
+        )
+        self.assertEqual(
+            move.line_ids.filtered(
+                lambda line: line.account_id == self.tax_received_account).tax_tag_ids,
+            tax_20_tag_minus
+        )
```

Comment:
I think this part isn't required because it just ensure that everything is working correctly in normal circumstances which is already done by standard tests

Reply:
As you pointed out, what creates the issue is that there is a `pos.order`/`pos_session` that is opened.
When there is one open, the test you wrote ensures it's working correctly.
When there isn't any, the situation is the same as if `point_of_sale` wasn't installed and thus is already covered by the standard test in `account`.

I think it's not required but if you prefer to keep it, I leave the final decision on that to someone from the PoS team.
They probably won't mind to keep it.